### PR TITLE
Implements ensembling of models

### PR DESCRIPTION
**Pull request by ftoralesacosta**
Ensembling is implemented in RunModel. This reduces parallelization but results in overall less variance. It’s called ‘step ensembling’ since the ensembling is done within each step, before passing onto the next step or iteration. An alternative would be to call a script and run the OmniFold procedure as a whole (for all iterations), [n_ensemble] times.

Run ensembling with new class variable: self.n_ensemble

The implementation initializes lists, to hold the models.
self.step1_models = []
self.step2_models = []

The models are saved, so that subsequent iterations can be start training from the previous iteration.

CompileModel was changed to take any single model as input, with CompileModel*s* reproduces the original function.

The reweigh function now averages the output of all the models, when self.n_ensembles is larger than 1

### CODE REVIEW

File: omnifold/omnifold.py [hunk 1]
```diff
@@ -178,13 +179,16 @@ def RunStep1(self,i):
         #Don't update weights where there's no reco events
         new_weights = np.ones_like(self.weights_pull)
         new_weights[self.mc.pass_reco] = self.reweight(self.mc.reco,self.model1,batch_size=1000)[self.mc.pass_reco]
+        # new_weights[self.mc.pass_reco] = self.reweight(self.mc.reco,self.step1_models[0],batch_size=1000)[self.mc.pass_reco]
+        print(self.weights_pull)
         self.weights_pull = self.weights_push *new_weights
+        print(self.weights_pull)
```
Comment (ViniciusMikuni) [inline omnifold/omnifold.py:185]: maybe take the print off

File: omnifold/omnifold.py [hunk 2]
```diff
@@ -194,7 +198,10 @@ def RunStep2(self,i):
         )
         new_weights = np.ones_like(self.weights_push)
         new_weights[self.mc.pass_gen]=self.reweight(self.mc.gen,self.model2)[self.mc.pass_gen]
+        # new_weights[self.mc.pass_gen]=self.reweight(self.mc.gen,self.step2_models[0])[self.mc.pass_gen]
+        print(self.weights_push)
         self.weights_push = new_weights
+        print(self.weights_push)
```
Comment (ViniciusMikuni) [inline omnifold/omnifold.py:204]: same here

**Review by ViniciusMikuni — CHANGES_REQUESTED** [review]
Just a few comments to take out some print statements in the main class

File: omnifold/omnifold.py (hunk 2 above)
Comment (ftoralesacosta) [reply omnifold/omnifold.py:204]: Done. Also took out the print's in CompielModels.

**Pull request merged**